### PR TITLE
Fix s1_geocode_metadata crash

### LINKED ISSUE #110: [Bug]: geocode metadata fails
### Checked for duplicates

Yes - I've already checked

### Describe the bug

When generating gslcs with geocoded metadata get the following error:

```python traceback
journal: s1_rdr2geo burst successfully ran in 0:04:09 (hr:min:sec)
journal: Starting s1_geocode_metadata burst
Traceback (most recent call last):
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_cslc.py", line 54, in <module>
    main()
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_cslc.py", line 49, in main
    run(parser.run_config_path, parser.args.grid_type)
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_cslc.py", line 43, in run
    s1_geocode_slc.run(cfg)
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_geocode_slc.py", line 102, in run
    s1_geocode_metadata.run(cfg, burst, fetch_from_scratch=True)
  File "/mnt/aurora-r0/cabrera/software/tools/COMPASS/src/compass/s1_geocode_metadata.py", line 73, in run
    geo.lines_per_block = blocksize
AttributeError: 'isce3.ext.isce3.geocode.GeocodeFloat32' object has no attribute 'lines_per_block'
```

### What did you expect?

I expected geocoded metadata

### Reproducible steps

```bash
1. A sample config file for one burst can be found from 
2. /mnt/aurora-r0/cabrera/data/compass_test/stack/runconfigs/geo_runconfig_20210725_t013_026557_iw2.yaml
3. ran by doing ./stack/run_files/run_20210725_t013_026557_iw2.sh
...
```

### Environment

```bash
- Using COMPASS 0.1.3
- isce3 0.10.0 from conda
...
```

### PR DESCRIPTION
**Pull request by LiangJYu**
This PR fixes #110 
- fix s1_geocode_metadata incorrect usage of block params
- enable s1_geocode_metadata in unit tests

### CODE REVIEW

File: src/compass/s1_geocode_metadata.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def run(cfg, burst, fetch_from_scratch=False):
     geo.doppler = image_grid_doppler
     geo.threshold_geo2rdr = threshold
     geo.numiter_geo2rdr = iters
-    geo.lines_per_block = blocksize
+    elements_in_block = lines_per_block * geogrid.width
```
Comment (scottstanie) [inline src/compass/s1_geocode_metadata.py:73]:
```suggestion
    elements_in_block = lines_per_block * geo)grid.width
```


```
journal: Starting s1_geocode_metadata burst
Traceback (most recent call last):
  File "/u/aurora-r0/staniewi/miniconda3/envs/mapping/bin/s1_cslc.py", line 8, in <module>
    sys.exit(main())
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_cslc.py", line 49, in main
    run(parser.run_config_path, parser.args.grid_type)
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_cslc.py", line 43, in run
    s1_geocode_slc.run(cfg)
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_geocode_slc.py", line 104, in run
    s1_geocode_metadata.run(cfg, burst, fetch_from_scratch=True)
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_geocode_metadata.py", line 73, in run
    elements_in_block = lines_per_block * geogrid.width
NameError: name 'geogrid' is not defined. Did you mean: 'geo_grid'?
```

**Comment by scottstanie** [comment]
I think it's unrelated to your PR, this is also erroring:

```
Traceback (most recent call last):
  File "/u/aurora-r0/staniewi/miniconda3/envs/mapping/bin/s1_cslc.py", line 8, in <module>
    sys.exit(main())
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_cslc.py", line 49, in main
    run(parser.run_config_path, parser.args.grid_type)
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_cslc.py", line 43, in run
    s1_geocode_slc.run(cfg)
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/s1_geocode_slc.py", line 191, in run
    corrections_to_h5group(cslc_group, burst, cfg, rg_lut, az_lut, scratch_path,
  File "/u/aurora-r0/staniewi/repos/COMPASS/src/compass/utils/h5_helpers.py", line 661, in corrections_to_h5group
    extended_coeffs_group = correction_group.require_group('extended_coefficients')
UnboundLocalError: local variable 'correction_group' referenced before assignment
```

**Comment by scottstanie** [comment]
And the CI build is failing because it might be taking too long? 
https://app.circleci.com/pipelines/github/opera-adt/COMPASS/289/workflows/14e1cfca-610a-47ac-8c93-5b6652683b6e/jobs/285

```
#!/bin/bash -eo pipefail
docker run test_image sh -c "cd ~/OPERA/COMPASS && pytest"

============================= test session starts ==============================
platform linux -- Python 3.11.0, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/compass_user/OPERA/COMPASS
collected 4 items

tests/test_ionex.py ..                                                   [ 50%]
tests/test_s1_geocode_slc.py 
CondaError: KeyboardInterrupt


Too long with no output (exceeded 10m0s): context deadline exceeded
```

**Comment by LiangJYu** [comment]
> And the CI build is failing because it might be taking too long? https://app.circleci.com/pipelines/github/opera-adt/COMPASS/289/workflows/14e1cfca-610a-47ac-8c93-5b6652683b6e/jobs/285
> 
> ```
> #!/bin/bash -eo pipefail
> docker run test_image sh -c "cd ~/OPERA/COMPASS && pytest"
> 
> ============================= test session starts ==============================
> platform linux -- Python 3.11.0, pytest-7.2.2, pluggy-1.0.0
> rootdir: /home/compass_user/OPERA/COMPASS
> collected 4 items
> 
> tests/test_ionex.py ..                                                   [ 50%]
> tests/test_s1_geocode_slc.py 
> CondaError: KeyboardInterrupt
> 
> 
> Too long with no output (exceeded 10m0s): context deadline exceeded
> ```

Yup. I'm testing to see if if's faster with layover shadow turned off.

We can also increase the allowed test runtime.

**Comment by scottstanie** [comment]
Another possible unit test fix: `pytest --capture=tee-sys` would print out the stderr /stdout stuff while the test is running. it looks like the problem is nothing's getting printed (since pytest captures all by default while the test is running)

**Review by scottstanie — APPROVED** [review]

**Review by taliboliver — APPROVED** [review]
Thank you @LiangJYu  this worked for me and fixed the issue #110

**Pull request merged**